lib/pages/company_site/eti/table_products.rb

```diff
@@ -0,0 +1,459 @@
+        end
+
+        # @return nothing
+        #
+        # @example
+        #   set_description(@product1, 'Полное описание товара или услуги')
+        #
+        # @param [Selenium::WebDriver::Element]
+        # @param [String]
+        #
+        def set_description(product, text)
+          click_on_cell(descriptions_elements[product_index(product)])
+
+          browser
+            .action
+            .send_keys(Selenium::WebDriver::Keys::KEYS[:enter])
+            .perform
+
+          description_popup = DescriptionPopup.new
+          description_popup.wait_for_visible
+          description_popup.text_element.click
+
+          browser
+            .action
+            .key_down(:control)
+            .send_keys('a')
+            .key_up(:control)
+            .send_keys(Selenium::WebDriver::Keys::KEYS[:clear])
+            .perform
+
+          description_popup.text_element.send_keys(text)
+          description_popup.save
+
+          wait_saving
+        end
+
+        # @return nothing
+        #
+        # @example
+        #   set_group(@product1, 'Спортивный инвентарь')
+        #
+        # @param [Selenium::WebDriver::Element]
+        # @param [String]
+        #
+        def set_group(product, group)
+          click_on_cell(groups_elements[product_index(product)])
+
+          groups_binding_popup = GroupsBindingPopup.new
+          groups_binding_popup.wait_for_visible
+          groups_binding_popup.select_group(group)
+
+          wait_saving
+        end
+
+        def group(product)
+          groups_elements[product_index(product)].element.text.strip
+        end
+
+        # @return nothing
+        #
+        # @example
+        #   upload_image(@product1, type: :local, path: 'C:/images/picture.png')
+        #   upload_image(@product1, type: :remote, url: 'http://site.ru/images/picture.png')
+        #
+        # @param [Selenium::WebDriver::Element]
+        # @param [Hash] - тип источника
+        #   * :remote - удаленный сервер (http/https протокол)
+        #   * :local - локальная файловая система
+        # @param [Hash]
+        #  * :url - ссылка на изображение (при type: :remote)
+        #  * :path - путь к файлу (при type: :local)
+        #
+        def upload_image(product, options = {})
+          click_on_cell(images_elements[product_index(product)])
+
+          images_upload_popup = ImagesUploadPopup.new
+          images_upload_popup.wait_for_visible
+
+          case options[:type].to_sym
+          when :remote
+            images_upload_popup.upload_from_url(options[:url])
+          when :local
+            images_upload_popup.upload_from_file(options[:path])
+          end
+
+          images = images_upload_popup.images
+          images_upload_popup.close
+          images
+        end
+
+        def image(product)
+          {
+            url: images_elements[product_index(product)].attribute('src'),
+            counts: images_counters_elements[product_index(product)].text.to_i
```

Comment:
Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.